stan/math/opencl/qr_decomposition.hpp

```diff
@@ -0,0 +1,110 @@
+#ifndef STAN_MATH_OPENCL_QR_DECOMPOSITION_HPP
+#define STAN_MATH_OPENCL_QR_DECOMPOSITION_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/prim/fun/constants.hpp>
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/zeros_strict_tri.hpp>
+#include <stan/math/opencl/prim/identity_matrix.hpp>
+#include <stan/math/opencl/kernel_generator.hpp>
+#include <stan/math/opencl/prim/multiply.hpp>
+#include <stan/math/opencl/prim/sum.hpp>
+#include <stan/math/opencl/copy.hpp>
+#include <CL/cl.hpp>
+#include <algorithm>
+#include <vector>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+/**
+ * Calculates QR decomposition of A using the block Householder algorithm.
```

Comment:
might add link to description of block Householder on GPUs

Reply:
You mean the article I followed when implementing this?

Reply:
Sure